experimental/experimental_examples/cpp_api/taskgraph_filtering.cc

```diff
@@ -0,0 +1,288 @@
+  (void)o;
+  (void)c;
+
+  for (uint32_t r = 0; r < array_size; ++r) {
+    assert(expected_results[r] == data->at(r));
+  }
+}
+
+auto compress_chunk(const input_info<uint32_t>& in) {
+  auto [offset, chunk_size, data_ptr] = in;
+
+  unsigned int out_size = 2 * data_ptr->size() * sizeof(uint32_t);
+  std::vector<char> compressed_data(out_size);
+  int rc = BZ2_bzBuffToBuffCompress(
+      compressed_data.data(),
+      &out_size,
+      reinterpret_cast<char*>(data_ptr->data()),
+      data_ptr->size() * sizeof(uint32_t),
+      1,
+      0,
+      0);
+
+  if (rc != BZ_OK) {
+    throw std::runtime_error("compression error");
+  }
+
+  compressed_data.resize(out_size);
+
+  return std::make_tuple(
+      offset, chunk_size, std::make_shared<std::vector<char>>(compressed_data));
+}
+
+auto decompress_chunk(const input_info<char>& in) {
+  auto [offset, chunk_size, data_ptr] = in;
+
+  unsigned int out_size = chunk_size * sizeof(uint32_t);
+  std::vector<uint32_t> decompressed_data(chunk_size);
+
+  int rc = BZ2_bzBuffToBuffDecompress(
+      reinterpret_cast<char*>(decompressed_data.data()),
+      &out_size,
+      data_ptr->data(),
+      static_cast<unsigned int>(data_ptr->size()),
+      0,
+      0);
+  if (rc != BZ_OK) {
+    throw std::runtime_error("decompression error");
+  }
+
+  assert(out_size / sizeof(uint32_t) == chunk_size);
+
+  return std::make_tuple(
+      offset,
+      chunk_size,
+      std::make_shared<std::vector<uint32_t>>(decompressed_data));
+}
+
+int main() {
+  Config cfg;
+  Context ctx(cfg);
+  std::string array_name("taskgraph_filtering");
+  std::string output_array("taskgraph_filtering_output");
+  uint32_t array_size = 100;
+
+  if (Object::object(ctx, array_name).type() == Object::Type::Array) {
+    tiledb::VFS vfs(ctx);
+    vfs.remove_dir(array_name);
+    std::cout << "Removed existing array" << std::endl;
+  }
+  if (Object::object(ctx, output_array).type() == Object::Type::Array) {
+    tiledb::VFS vfs(ctx);
+    vfs.remove_dir(output_array);
+    std::cout << "Removed existing output array" << std::endl;
+  }
+
+  create_array(ctx, output_array);
+  create_array(ctx, array_name);
+  generate_test_data(ctx, array_name, array_size);
+
+  uint32_t num_threads = stoi(cfg.get("sm.compute_concurrency_level"));
+  uint32_t chunk_size = array_size / num_threads;
+
+  chunk_generator gen{array_size, chunk_size};
+  gen.reset();
+
+  TaskGraph<DuffsScheduler<node>> graph(num_threads);
+
+  for (uint32_t w = 0; w < num_threads; ++w) {
+    auto a = initial_node(graph, gen);
+    auto b = transform_node(
+        graph,
+        [&ctx, &array_name, chunk_size, array_size](uint32_t in_offset_begin) {
+          return read_chunk(
+              ctx, array_name, chunk_size, in_offset_begin, array_size);
+        });
+    auto c = transform_node(graph, compress_chunk);
+
+    // TODO: MIMO node impl doesn't seem available, so I'm not sure how to
```

Comment:
Clarify TODO's before merging.

Reply:
Requires SC-25436. Let's mention the story in the comment, and we can update it (or create another variant) when that is available.